### ISSUE TITLE
owncloud syncs edited photos endless times

### ISSUE THREAD

**Issue by Obi-Wahn** [owncloud/android#1828]
### Actual behaviour

-Tell us what happens

Photos which were edited with the Google Photos app  are synced again and again. But only the edited photo is synced multiple times. (IMG_20161018_112856.jpg, IMG_20161018_112856 (2).jpg, IMG_20161018_112856 (3).jpg, .....) at least it goes to 10.... deleting the false files doesn't help. 
### Expected behaviour

-Tell us what should happen

Only one edited photos should be uploaded or the original photo should be replaced.
### Steps to reproduce
1. Take a Photo with Google Camera
2. Edit the photo (make it black/white)
3. Wait for the sync on the pc.
4. Android device and pc should be active and with enabled sync.

Can this problem be reproduced with the official owncloud server?
(url: https://demo.owncloud.org, user: test, password: test)
## 
### Environment data

Android version: 7.0

Device model: Nexus 5x

Stock or customized system: Stock

ownCloud app version: 2.1.2

ownCloud server version: 9.1.1
### Logs

**.owncloudsync.log:**
```
|0|SofortUpload|INST_SYNC|Down|1476816713|58066f49a0ca5|0|00001117ocns41bl4o7x|4||0|0|1476797137|||INST_NONE|
18:52:10|411|SofortUpload/IMG_20160923_111020 (2).jpg|INST_NEW|Down|1476816713|6554ee207cd679cce321b2f79e8a4b0a|681078|00015993ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476816841|58066fc9ea895|0|00001117ocns41bl4o7x|4||0|0|1476816726|||INST_NONE|
18:54:06|645|SofortUpload/IMG_20160930_123007 (2).jpg|INST_NEW|Down|1476816841|e88f1afe1a4cbf7138884232da8e62f5|1550469|00015998ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476816908|5806700d047ce|0|00001117ocns41bl4o7x|4||0|0|1476816841|||INST_NONE|
18:55:36|345|SofortUpload/IMG_20160921_093610 (2).jpg|INST_NEW|Down|1476816908|e3ec060704b55ad42907414ebba78e47|937415|00016001ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476816955|5806703bbd7af|0|00001117ocns41bl4o7x|4||0|0|1476816931|||INST_NONE|
18:56:36|299|SofortUpload/IMG_20160808_132013 (2).jpg|INST_NEW|Down|1476816955|58ff45cd74113c2192aad97f93ebbc3e|862507|00016004ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476817041|58067091d00cb|0|00001117ocns41bl4o7x|4||0|0|1476816991|||INST_NONE|
18:57:36|424|SofortUpload/IMG_20161018_112856 (2).jpg|INST_NEW|Down|1476817009|58ba443179f20628fe102ab4d9cd716a|939133|00016007ocns41bl4o7x|4||0|0|0|||INST_NONE|
18:57:36|573|SofortUpload/IMG_20161018_112908 (2).jpg|INST_NEW|Down|1476817041|d867a9a4476e791b4e65930ef5c95fa7|598093|00016010ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476884986|580779faea3f4|0|00001117ocns41bl4o7x|4||0|0|1476817052|||INST_NONE|
13:52:32|610|SofortUpload/IMG_20160923_111020 (3).jpg|INST_NEW|Down|1476884986|4ba2841e208ca4246b24d4d1c4fec069|681078|00016052ocns41bl4o7x|4||0|0|0|||INST_NONE|
13:52:32|1323|SofortUpload/IMG_20161018_204731.jpg|INST_NEW|Down|1476822166|113c56f053ae7a6fd35d125b13bfd7b8|3992239|00016036ocns41bl4o7x|4||0|0|0|||INST_NONE|
13:52:33|20279|SofortUpload/IMG_20161018_204743.jpg|INST_NEW|Down|1476822160|34eba5349df0813f04177fc2b5107912|3718639|00016029ocns41bl4o7x|4||0|0|0|||INST_NONE|
13:52:33|3825|SofortUpload/IMG_20161018_204757.jpg|INST_NEW|Down|1476822152|e9353b5b0fa830debd8a6d5c67efee6e|2525961|00016022ocns41bl4o7x|4||0|0|0|||INST_NONE|
13:52:34|19459|SofortUpload/IMG_20161018_204819.jpg|INST_NEW|Down|1476822143|781f472983bf2715965a2b3df99f3bd6|3010826|00016016ocns41bl4o7x|4||0|0|0|||INST_NONE|
13:52:37|5038|SofortUpload/IMG_20161019_093452.jpg|INST_NEW|Down|1476884864|1edafab8a6560e3c1d14cc15d53562eb|2726950|00016049ocns41bl4o7x|4||0|0|0|||INST_NONE|
13:52:42|1395|SofortUpload/IMG_20161019_112955.jpg|INST_NEW|Down|1476884851|a3f4e38723da8445cc700cf7dce45649|3237204|00016043ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476886763|580780eb58d96|0|00001117ocns41bl4o7x|4||0|0|1476885168|||INST_NONE|
14:19:42|472|SofortUpload/IMG_20160930_123007 (3).jpg|INST_NEW|Down|1476886763|9b2e39405da144eeff5b06a3902328a4|1550469|00016057ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476886795|5807810c1228f|0|00001117ocns41bl4o7x|4||0|0|1476886777|||INST_NONE|
14:20:41|499|SofortUpload/IMG_20160921_093610 (3).jpg|INST_NEW|Down|1476886795|8002c88baa73ce455decb46db6636b35|937415|00016060ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476886860|5807814c4f33e|0|00001117ocns41bl4o7x|4||0|0|1476886836|||INST_NONE|
14:21:41|664|SofortUpload/IMG_20160808_132013 (3).jpg|INST_NEW|Down|1476886859|82ff76bd01257aa81a92ff4f06191f1e|862507|00016063ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476887040|5807820103c0a|0|00001117ocns41bl4o7x|4||0|0|1476886896|||INST_NONE|
14:24:11|381|SofortUpload/IMG_20161018_112856 (3).jpg|INST_NEW|Down|1476887040|151919f279cb99c0940fcddd27468435|939133|00016066ocns41bl4o7x|4||0|0|0|||INST_NONE|
# =#=#=# Syncrun started  until  (0 msec)

|0|SofortUpload|INST_SYNC|Down|1476887142|580782668c2c8|0|00001117ocns41bl4o7x|4||0|0|1476887046|||INST_NONE|
14:26:11|315|SofortUpload/IMG_20161018_112908 (3).jpg|INST_NEW|Down|1476887142|af1d59156137157505404b53bd12bdb4|598093|00016069ocns41bl4o7x|4||0|0|0|||INST_NONE|

```

**Comment by davivel** [comment]
Hi, @Obi-Wahn .

Do you mean that after editing the photo just once with Google Photos it is uploaded several times to the account?

**Comment by Obi-Wahn** [comment]
Yes. I think there is a connection between editing the old photo and opening the owncloud app.
I just edited a new photo and right after i opened the owncloud app three (old) other photos were reuploaded the sixth time.

**Comment by davivel** [comment]
To decide when a new picture was taken, the app watches for file events in the camera folder. There are several sequences of file events considered as meaning "new picture there!", because we detected that different camera apps handle the dump of the picture in different ways. Since the editor writes the file in the same folder, it generates file events that 

To fix this we need to drop the current solution for instant uploads based in file observers to another based in regular scanning of the camera folder. There are other advantages in this approach (and some drawbacks, of course). Probably we will do this in some moment, but I'm afraid it will not be in the short term.

Alternatives meanwhile? In the instant upload settings, change the default value of ```Original file will be...```  from ```kept in original folder``` to ```moved to app folder```. The pictures will **moved** to the owncloud folder, and you can edit them there without duplications.

**Comment by davigonz** [comment]
Hi @Obi-Wahn , thanks for your feedback.

We have finally drop the Instant uploads feature and replaced it with Camera uploads feature, which is going to be included in the next Android app release (2.6.0).

To use it, you just have to select a camera folder and the pictures and videos contained there will be automatically uploaded to the folder you select in a period of 15 minutes at most.

In your case, you shouldn't see duplicated uploads with the new feature since we are not watching for file events any more. 

The feature is already available in 2.5.1 beta version in PlayStore or F-Droid, so you can test it whenever you want and tell us if works for you.
